clean/ca/ventura_county_sheriff.py

```diff
@@ -0,0 +1,219 @@
+        # Start page contains list of "detail"/child pages with links to the SB16/SB1421/AB748
+        #  videos and files along with additional index pages
+        self.base_url = "https://www.venturasheriff.org"
+        self.index_urls = {
+            f"{self.base_url}/sb1421/officer-involved-shooting-ois/": "ois.html",
+            f"{self.base_url}/sb1421/use-of-force-great-bodily-injury-cases-gbi/": "gbi.html",
+            f"{self.base_url}/ab748/": "ab748.html",
+        }
+
+        self.cache = Cache(cache_dir)  # ~/.clean-scraper/cache/
+        self.data_dir = data_dir
+        self.cache_dir = cache_dir
+
+        # Use module path to construct agency slug, which we'll use downstream
+        # to create a subdir inside the main cache directory to stash files for this agency
+        mod = Path(__file__)
+        state_postal = mod.parent.stem
+        self.cache_suffix = f"{state_postal}_{mod.stem}"  # ca_ventura_county_sheriff
+        self.cache_root = cache_dir / (self.cache_suffix)
+        self.subpages_dir = self.cache_root / "subpages"
+
+    def scrape_meta(self, throttle: int = 0) -> Path:
+        metadata: List[MetadataDict] = []
+        page_urls = []
+        # Scrape index pages for both assets and links to case directories/subpages
+        for index_url in self.index_urls:
+            detail_page_links, local_metadata = self._process_index_page(index_url)
+            page_urls.extend(detail_page_links)
+            metadata.extend(local_metadata)
+            time.sleep(throttle)
+
+        # Now, process the links of case directories/subpages
+        for page_url in page_urls:
+            local_metadata = self._process_detail_page(page_url)
+            metadata.extend(local_metadata)
+            time.sleep(throttle)
+
+        outfile = self.data_dir.joinpath(f"{self.cache_suffix}.json")
+        logger.debug(f"Attempting to save metadata to {outfile}")
+        full_filename = self.cache.write_json(outfile, metadata)
+        return full_filename
+
+    # Helper/Private Methods
+    def _process_detail_page(self, target_url) -> List[MetadataDict]:
+        """Extract links to files such as videos from a detail page and write to JSON file."""
+        local_metadata: List[MetadataDict] = []
+
+        # Build a complete URL and determine the subdirectory name
+        if target_url.endswith("/"):
+            target_url = target_url[:-1]
+        if "http" not in target_url:
+            target_url = urllib.parse.urljoin(self.base_url, target_url)
+
+        full_filename = self.subpages_dir / (target_url.split("/")[-1] + ".html")
+        relative_filename = str(full_filename.relative_to(self.cache_dir)).replace(
+            "\\", "/"
+        )
+
+        # Download the index page, which saves to local cache
+        self.cache.download(
+            full_filename,
+            target_url,
+            force=False,  # Do NOT automatically rescrape subpages
```

Comment:
Could you walk me through setting it to `False` here? On the backend side, once these are saved to a local cache, they're copied over to Azure Blob Storage with timestamps. Leaving it as `True` ensures we're catching changes to the page and its asset_url's byte lengths

Reply:
Whatever you'd like to do is fine by me. Looks like there were about 58 subpages and I was assuming there was no real reason to rescrape those. Index pages are of course forced to rescrape.

Reply:
There's no info on file size here that I'm aware of, FYI. So unless you rescrape the asset itself; or alternately grab the HEAD and get accurate, reliable results; rescraping the subpage would only show you any new assets, and I don't know if anything ever gets refreshed.